bubblescan-client/src/App.tsx

```diff
@@ -48,59 +49,180 @@ function App() {
+                text-align: center;
+                break-inside: avoid;
+                margin-bottom: 2px;
+                font-size: 7px;
+                width: 100%;
+                box-sizing: border-box;
+              }
+              .bubble {
+                width: 5px;
+                height: 5px;
+                border-radius: 50%;
+                border: 1px solid black;
+                display: inline-flex;
+                align-items: center;
+                justify-content: center;
+                font-size: 6px;
+                text-align: center;
+                margin: 1px;
+              }
+              .options {
+                display: flex;
+                justify-content: center;
+                gap: 1px;
+                margin-top: 1px;
+                align-items: center;
+              }
+              .option-label {
+                margin-right: 1px;
+                font-weight: bold;
+                font-size: 6px;
+              }
+              .right-column {
+                border-left: 1px solid #ddd;
+                padding-left: 10px;
+              }
+              .id-container {
+                margin-top: 20px;
+              }
+              .id-header {
+                font-weight: bold;
+                text-align: center;
+                font-size: 9px;
+                margin-bottom: 10px;
+              }
+              .id-row {
+                display: flex;
+                justify-content: space-around;
+                margin-bottom: 10px;
+              }
+              .id-column {
+                display: flex;
+                flex-direction: column;
+                align-items: center;
+              }
+              .id-bubble {
+                width: 10px; /* Increased size for the student ID bubbles */
+                height: 10px; /* Increased size for the student ID bubbles */
                 border-radius: 50%;
-                display: inline-block;
+                border: 1px solid black;
+                display: inline-flex;
+                align-items: center;
+                justify-content: center;
+                font-size: 6px; /* Keep the font size same for readability */
+                text-align: center;
+                margin: 2px; /* Slightly larger margin */
+              }
+              .key-id-container {
+                margin-top: 20px;
+                border: 1px solid #ddd;
+                padding: 10px;
+                text-align: center;
+                border-radius: 5px;
+                width: 120px;
+                margin-left: auto;
+                margin-right: auto;
+              }
+              .key-id-header {
+                font-weight: bold;
+                font-size: 8px;
+                margin-bottom: 5px;
+              }
+              .key-id-options {
+                display: flex;
+                justify-content: space-between;
+                gap: 5px;
```

Comment:
You should move the styling into a .css file

Reply:
html styling was needed in the generateExam function